app/models/user.rb

```diff
@@ -1,12 +1,19 @@
 # frozen_string_literal: true
 
 class User < ApplicationRecord
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  before_validation { self.email = email.downcase }
```

Comment:
It's better mo move callbacks after the validations according to the style guide https://github.com/arbox/rails-style-guide/blob/master/README-ruRU.md#%D0%9C%D0%BE%D0%B4%D0%B5%D0%BB%D0%B8